File: org.dcs.core/src/main/java/org/dcs/core/module/flow/FlowModule.java
package org.dcs.core.module.flow;

import java.util.Map;
import java.util.Properties;

import org.dcs.api.service.RESTException;
import org.osgi.framework.BundleContext;

public interface FlowModule {
	
  public void init(BundleContext bundleContext);
  
	public Map<String, Properties> getPropertyDescriptors();
	
	public Map<String, Properties> getRelationships();
	
	public void schedule();
	
	public byte[] trigger(Properties properties) throws RESTException ;
	
	public void unschedule();
	
	public void stop();
	
	public void shutdown();
	
	public void remove();

}

File: org.dcs.core/src/main/scala/org/dcs/core/module/flow/FlowModule.scala
package org.dcs.core.module.flow

import java.util.{Map => JavaMap}

import org.osgi.framework.BundleContext


trait FlowModule {

  def init( bundleContext: BundleContext)
  
	def getPropertyDescriptors(): JavaMap[String, JavaMap[String, String]]
	
	def getRelationships(): JavaMap[String, JavaMap[String, String]]
	
	def schedule()
	
	def trigger(properties: JavaMap[String, String]): Array[Byte]
	
	def unschedule()
	
	def stop()
	
	def shutdown()
	
	def remove()

}
